superset/db_engine_specs/bigquery.py

```diff
@@ -86,6 +93,17 @@ class BigQueryEngineSpec(BaseEngineSpec):
         "P1Y": "{func}({col}, YEAR)",
     }
 
+    custom_errors = {
+        CONNECTION_DATABASE_PERMISSIONS_REGEX: (
+            __(
+                "We were unable to connect to your database. Please "
+                "confirm that your service account has the Viewer "
+                "and Job User roles on the project."
```

Comment:
This should also be simple, just describing the error. The actionable item should be in the issue code.

Reply:
This is the message sophie wanted specifically for user permissions